storage/flux/window.go

```diff
@@ -77,6 +82,7 @@ func (w *windowTableSplitter) Do(f func(flux.Table) error) error {
 			table := &windowTableRow{
 				buffer: buffer,
 				done:   done,
+				empty:  w.createEmpty && values.IsNull(i),
```

Comment:
Instead of doing this, can you move this check to line 76 above so we drop the null values when we're using a selector?

```
if w.selector && values.IsValid(i) {
    for j, arr := range arrs {
        buffer.Values[j] = arrow.Slice(arr, int64(i))
    }
} else {
    for j, arr := range arrs {
        // might want to add a utility method for creating an empty array into the arrow package
        buffer.Values[j] = arrow.NewBuilder(cr.Cols()[j].Type, w.alloc).NewArray()
    }
}
```